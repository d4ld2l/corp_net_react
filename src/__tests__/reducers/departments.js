import reducer from '../../redux/reducers/departments'

describe('departments reducer', () => {
  const initialState = []

  it('should match GET_DEPARTMENTS_TREE_RES snapshot', () => {
    const action = {
      type: 'GET_DEPARTMENTS_TREE_RES',
      payload: [
        {
          id: 2,
          name: 'Университет',
          code: 'Университет',
          uuid: 'Университет',
          company_id: null,
          created_at: '2017-08-21T13:36:26.676Z',
          updated_at: '2017-10-05T14:38:34.649Z',
          logo: {},
          departments_tree: [
            {
              id: 2,
              company_id: 3,
              code: 'root',
              name_ru: 'Допинг-контроль',
              parent_id: null,
              region: 36,
              created_at: '2017-08-04T14:20:10.796Z',
              updated_at: '2017-10-05T14:51:58.167Z',
              legal_unit_id: 1,
              manager_id: null,
              logo: {},
              children: [],
              participants: [],
              participants_count: 1,
            },
          ],
        },
      ],
    }

    expect(reducer(initialState, action)).toMatchSnapshot()
  })
})
