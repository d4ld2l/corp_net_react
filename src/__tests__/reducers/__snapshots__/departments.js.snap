// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`departments reducer should match GET_DEPARTMENTS_TREE_RES snapshot 1`] = `
Array [
  Object {
    "code": "Университет",
    "company_id": null,
    "created_at": "2017-08-21T13:36:26.676Z",
    "departments_tree": Array [
      Object {
        "children": Array [],
        "code": "root",
        "company_id": 3,
        "created_at": "2017-08-04T14:20:10.796Z",
        "id": 2,
        "legal_unit_id": 1,
        "logo": Object {},
        "manager_id": null,
        "name_ru": "Допинг-контроль",
        "parent_id": null,
        "participants": Array [],
        "participants_count": 1,
        "region": 36,
        "updated_at": "2017-10-05T14:51:58.167Z",
      },
    ],
    "id": 2,
    "logo": Object {},
    "name": "Университет",
    "updated_at": "2017-10-05T14:38:34.649Z",
    "uuid": "Университет",
  },
]
`;
